test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+﻿using NetTopologySuite.Densify;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.Geometries.Utilities;
+using NetTopologySuite.IO;
+using NetTopologySuite.Operation.Polygonize;
+using NetTopologySuite.Operation.Valid;
+using NetTopologySuite.Precision;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace NetTopologySuite.Samples.Technique
+{
+    /// <summary>
+    /// Presents a technique for converting coordinates to a polygon. With the coordinates all in one hemisphere it is a fairly simple process.
+    /// However, when the geometry crosses the dateline it becomes a more challenging problem. This routine attempts to do both. That is to say,
+    /// it converts coordinates to a poly geometry regardless if the geometry traverses the dateline or not. If the geometry does cross the dateline
+    /// then it is split up into multiple parts. This benefits the user in two ways: first, any GIS will be able to display the geometry in a proper
+    /// way regardless of map projection and two, the GIS will be able to perform geoprocessing functions on the geometry. Note that standard feature
+    /// classes do not support individual features with multiple parts that share the same edge. Thus, it is left up to the user on how to commit
+    /// the geometries to a feature class. Suggestions are to save the multiple geometries as individual features or to slightly alter the edges of
+    /// each geometry so as to save all parts as a single multi-part feature. For the latter option, the algorithm provides an input to trim edges
+    /// of parts so that they can be used in a single feature.
+    /// </summary>
+    public class SplitPolygonAtDateline
+    {
+        [STAThread]
+        public static void main()
```

Comment:
maybe this code can be moved to a unit test (along with other sample code, possibly?) I know that other "Technique" items uses this parrern (inherited from JTS code) but basically this means that this code isn't tested.

Reply:
I haven't done any unit testing in any of my projects so am a complete rookie at that. As you mentioned I just followed the pattern of testing in other routines in the Technique area.

Reply:
I will help you with that

Reply:
@sindizzy  added the [unit test code](https://github.com/sindizzy/NetTopologySuite/commit/ed8b5368436a496200b94f3b7c2e6e6df2c86e75), hope to not have done a mess.
This is just a sample, please feel free to add your own asserts and other checks.